src/kademlia/get_item.cpp

```diff
@@ -70,6 +70,14 @@ void get_item::got_data(bdecode_node const& v,
 		{
 			if (!m_data.assign(v, salt, seq, pk, sig))
 				return;
+
+			// For get_item, we should call callback when we get data.
+			// The data maybe is not the latest, but we can update it later.
+			// Waitting transaction timeout (15 seconds) is not efficient.
+			// For put_item, it put new values to DHT when done(), so this
+			// change shouldn't break anything since put_item callback
+			// almost doing nothing, just a notification.
+			m_data_callback(m_data);
```

Comment:
> does this cause the callback to be called more than once?
Yes. Actually the callback function is get_mutable_item_callback in dht_tracker.cpp, which will call session_impl::get_mutable_callback to post an alert (dht_mutable_item_alert). In my mind, an alert is similar to a message/event, if user has a handler to handle it, It shouldn't involve any issues or any difficulty.
But I realized that, with this patch, user maybe receive two exactly same alerts (got_data/done), which could confuse user. If got_data() has post a alert, then done() shouldn't post same alert again. 

> Speaking of which, don't you think it would be useful to know whether the call was "early" or "authoritative"? i.e. whether it's the first response or the most recent version of the data (i.e. the last response).
It will be useful in some cases, but the implementation might be a bit difficult. We even don't know which response is last one. We could call callback as "authoritative" in done(), but again, user might receive two exactly same alters, but marked as "early" and "authoritative". It will be weird in my point of view.